packages/graph/hash_graph/lib/graph/src/types/schema/property_type.rs

```diff
@@ -0,0 +1,727 @@
+use std::collections::{HashMap, HashSet};
+
+use serde::{Deserialize, Serialize};
+
+use crate::types::schema::{
+    Array, DataTypeReference, OneOf, PropertyTypeReference, Uri, Validate, ValidationError,
+    ValueOrArray,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(
+    try_from = "PropertyTypeObjectRepr",
+    tag = "type",
+    rename = "object",
```

Comment:
I don't really like how we're defining `"type": "object"` here. It feels very out of place and a bit too "magic"-y.

I think I understand the reasoning, but I'd be keen to hear about alternatives so we can at least rule them out consciously, and probably document the decision in the code.

Reply:
Improved this in d2fbcfecbd8ae0b5235b5e3ffdc96688edbc11cc and 7873c9e62381b84f9897aaeea2afee8821e0bad3

Reply:
I don't know really any alternatives. Somehow we want to ensure, that the schema has `"type" = "object"`.